File: app/src/main/java/com/example/liyayu/myapplication/HomeActivity.java
package com.example.liyayu.myapplication;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AlertDialog;
import android.support.v7.widget.AppCompatButton;
import android.support.v7.widget.AppCompatTextView;
import android.view.View;

import com.example.liyayu.myapplication.baseFramework.BaseKotlinActivity;
import com.example.liyayu.myapplication.demoViews.bluetoothPrinterDemo.BluetoothActivity;
import com.example.liyayu.myapplication.demoViews.coordinatorlayoutDemo.CoordinatorLayoutActivity;
import com.example.liyayu.myapplication.demoViews.fontDemo.FontActivity;
import com.example.liyayu.myapplication.demoViews.hotfixRobustDemo.RobustMainActivity;
import com.example.liyayu.myapplication.demoViews.imageViewTintDemo.TestImgTintActivity;
import com.example.liyayu.myapplication.demoViews.kalle_demo.KalleLoginActivity;
import com.example.liyayu.myapplication.demoViews.recycleDemo.RecycleActivity;
import com.example.liyayu.myapplication.demoViews.rippleDemo.Coloring;
import com.example.liyayu.myapplication.demoViews.rippleDemo.RippleActivity;
import com.example.liyayu.myapplication.demoViews.transitionDemo.TransitionMainActivity;
import com.example.liyayu.myapplication.util.NormalUtilsKt;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

import static com.example.liyayu.myapplication.jpush.JPushUtilsKt.setJpushAlias;

/**
 * Created by liyayu on 2018/3/1.
 */

public class HomeActivity extends BaseKotlinActivity {

    @BindView(R.id.text1)
    AppCompatTextView text1;
    @BindView(R.id.text2)
    AppCompatTextView text2;
    @BindView(R.id.button1)
    AppCompatButton button1;
    @BindView(R.id.button2)
    AppCompatButton button2;
    @BindView(R.id.button4)
    AppCompatButton button4;
    @BindView(R.id.button5)
    AppCompatButton button5;
    @BindView(R.id.button6)
    AppCompatButton button6;
    @BindView(R.id.button7)
    AppCompatButton button7;
    @BindView(R.id.button8)
    AppCompatButton button8;
    @BindView(R.id.button9)
    AppCompatButton button9;
    @BindView(R.id.button_end)
    AppCompatButton buttonEnd;
    Intent intent;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        ButterKnife.bind(this);
        setJpushAlias(this, NormalUtilsKt.getVersionName(this));
        Coloring.get().setViewRipple(text2, button1, button2, button4, button5, button6, button7, button8,button9, buttonEnd);
    }

    @OnClick({R.id.text1, R.id.text2, R.id.button1, R.id.button2, R.id.button4,
            R.id.button5, R.id.button6, R.id.button7, R.id.button8, R.id.button9, R.id.button10, R.id.button_end})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.text1:
                new AlertDialog.Builder(this)
                        .setTitle("Title")
                        .setMessage("This is message")
                        .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {

                            }
                        })
                        .setPositiveButton("Confirm", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {

                            }
                        })
                        .create()
                        .show();
                break;
            case R.id.button_end:
                intent = new Intent(this, MainActivity.class);
//                intent = new Intent(this, TestErr.class);
                startActivity(intent);
                break;
            case R.id.button1:
                intent = new Intent(this, RecycleActivity.class);
                startActivity(intent);
                break;
            case R.id.button2:
                intent = new Intent(this, RippleActivity.class);
                startActivity(intent);
                break;
            case R.id.button4:
                intent = new Intent(this, TransitionMainActivity.class);
                startActivity(intent);
                break;
            case R.id.button5:
                intent = new Intent(this, TestImgTintActivity.class);
                startActivity(intent);
                break;
            case R.id.button6:
                intent = new Intent(this, com.example.liyayu.myapplication.demoViews.bigImgDemo.MainActivity.class);
                startActivity(intent);
                break;
            case R.id.button7:
                intent = new Intent(this, CoordinatorLayoutActivity.class);
                startActivity(intent);
                break;
            case R.id.button8:
                startAct(RobustMainActivity.class);
                break;
            case R.id.button9:
                startAct(BluetoothActivity.class);
                break;
            case R.id.button10:
                startAct(KalleLoginActivity.class);
                break;
            case R.id.text2:
                intent = new Intent(this, FontActivity.class);
                startActivity(intent);
                break;
        }
    }
}

File: app/src/main/java/com/example/liyayu/myapplication/HomeActivity.kt
package com.example.liyayu.myapplication

import android.app.Activity
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.view.View
import com.example.liyayu.myapplication.baseFramework.BaseKotlinActivity
import com.example.liyayu.myapplication.baseFramework.BaseRvAdapter
import com.example.liyayu.myapplication.baseFramework.initRecycleLayoutManger
import com.example.liyayu.myapplication.demoViews.baseDemo.BaseDemoActivity
import com.example.liyayu.myapplication.demoViews.bluetoothPrinterDemo.BluetoothActivity
import com.example.liyayu.myapplication.demoViews.coordinatorlayoutDemo.CoordinatorLayoutActivity
import com.example.liyayu.myapplication.demoViews.easy_permissions_demo.EasyPermissionActivity
import com.example.liyayu.myapplication.demoViews.fontDemo.FontActivity
import com.example.liyayu.myapplication.demoViews.hotfixRobustDemo.RobustMainActivity
import com.example.liyayu.myapplication.demoViews.imageViewTintDemo.TestImgTintActivity
import com.example.liyayu.myapplication.demoViews.kalle_demo.KalleLoginActivity
import com.example.liyayu.myapplication.demoViews.loading_demo.TestLoadingActivity
import com.example.liyayu.myapplication.demoViews.recycleDemo.RecycleActivity
import com.example.liyayu.myapplication.demoViews.rippleDemo.Coloring
import com.example.liyayu.myapplication.demoViews.rippleDemo.RippleActivity
import com.example.liyayu.myapplication.demoViews.tools_calculate.CalculateForTools
import com.example.liyayu.myapplication.demoViews.transitionDemo.TransitionMainActivity
import com.example.liyayu.myapplication.jpush.setJpushAlias
import com.example.liyayu.myapplication.util.getVersionName
import kotlinx.android.synthetic.main.activity_home.*
import kotlinx.android.synthetic.main.item_main.view.*
import org.jetbrains.anko.sdk25.coroutines.onClick

/**
 * Created by liyayu on 2018/11/8.
 * 首页
 */

//@TestInject(BaseDemoActivity::class)
class HomeActivity : BaseKotlinActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        setJpushAlias(this, getVersionName(this))

        setStatusColor(R.color.y_bg)
//        text1.onClick {
//            AlertDialog.Builder(this@HomeActivity)
//                    .setTitle("Title")
//                    .setMessage("This is message")
//                    .setNegativeButton("Cancel") { dialog, which -> }
//                    .setPositiveButton("Confirm") { dialog, which -> }
//                    .create()
//                    .show()
//        }
//        text2.onClick { startAct(FontActivity()) }
//        button_end.onClick { startAct(MainActivity()) }
        val dataArray = mutableListOf(
                GoItem("Loading框架", TestLoadingActivity()),
                GoItem("EasyPermission",EasyPermissionActivity()),
                GoItem("kalle网络框架",KalleLoginActivity()),
                GoItem("自定义控件集合",BaseDemoActivity()),
                GoItem("Robust热更新",RobustMainActivity()),
                GoItem("约束布局探索",CoordinatorLayoutActivity()),
                GoItem("大图模式",com.example.liyayu.myapplication.demoViews.bigImgDemo.MainActivity()),
                GoItem("利用tint属性修改颜色及展示方式",TestImgTintActivity()),
                GoItem("页面转场动画",TransitionMainActivity()),
                GoItem("布局点击涟漪效果",RippleActivity()),
                GoItem("recycleView中添加动画效果",RecycleActivity()),
                GoItem("蓝牙连接打印机demo",BluetoothActivity()),
                GoItem("门窗计算器",CalculateForTools()))

        val adapter = MainItem(dataArray, this)
        rv_ah_list.layoutManager = initRecycleLayoutManger(this)
        rv_ah_list.adapter = adapter
    }

    class MainItem(list: MutableList<GoItem>, val mActivity: BaseKotlinActivity) : BaseRvAdapter<GoItem>(list, R.layout.item_main) {
        override fun onBind(holderView: View, position: Int, data: GoItem) = with(holderView) {
            Coloring.get().setViewRipple(btn_im_show)
            btn_im_show.text = data.title
            btn_im_show.setOnClickListener {
                mActivity.startAct(data.activity ?: BaseDemoActivity())
            }
        }

    }

    data class GoItem(val title: String? = "", val activity: Activity? = BaseDemoActivity())
}